lib/ensureNano.js

```diff
@@ -0,0 +1,52 @@
+function ensureNano(nano, db) {
+    var nanoDb = nano.use(db);
+    return wrapNano(nano, nanoDb, db);
+}
+
+function wrapNano(nano, nanoDb, db) {
+    var nanoObject = {};
+    Object.keys(nanoDb).forEach(function(method) {
+        if (typeof nanoDb[method] == "function") {
+            return nanoObject[method] = function() {
+                ensureDatabase.call(this, nano, nanoDb, db, method, arguments)
+            };
+        }
+
+        if (typeof nanoDb[method] == "object") {
+            return nanoObject[method] = wrapNano(nano, nanoDb[method], db);
+        }
+
+        return nanoObject[method] = nanoDb[method];
```

Comment:
What does this do? (my ref: https://github.com/dscape/nano/blob/master/lib/nano.js#L665)